zenodo/modules/records/views.py

```diff
@@ -253,9 +253,13 @@ def pid_url(identifier, scheme=None, url_scheme='https'):
             scheme = idutils.detect_identifier_schemes(identifier)[0]
         except IndexError:
             scheme = None
-    if scheme and identifier:
-        return idutils.to_url(identifier, scheme, url_scheme=url_scheme)
-    return ""
+    try:
+        if scheme and identifier:
+            return idutils.to_url(identifier, scheme, url_scheme=url_scheme)
+    except Exception:
+        current_app.logger.info('URL generation for identifier {0} failed.'
```

Comment:
This originated from one specific error actually on Sentry, when there is a unicode character in some identifiers, specifically ['\u200b' zero-width-space](http://www.fileformat.info/info/unicode/char/200B/index.htm). We probably don't want to keep every malformed identifier error in sentry, since it's probably a user input issue...